### DISCUSSION TITLE
Difference in using migrations between v4.1 and v5

### DISCUSSION THREAD

**Discussion by zarkones** [adonisjs/core#1480] [Help] [answered]
Hello, I originally posted this on discord, but was instructed to leave it here. So I might as well take this chance to be more clear about my issue.

I'm migrating an API from 4.1 to 5. Now when writing migrations I've run into a few issues.

1)  Inside `up()` method I wasn't able to get `this.fn.now()` to work, in a matter of fact, I failed to obtain Knex reference at all.
From which variable may I obtain that reference to Knex variable?

2)  Using `on()` in context of `table.foreign(. . .).references(. . .).on(. . .)` I'm prompt with "_Property 'on' does not exist on type 'ReferencingColumnBuilder'_". 
Even trough I've found use example within 4.1 documentation, I failed to manage the same inside v5 docs.
Is the syntax now different or?

I hope that you find the issues clear, and if more information is needed, please don't hold your self from asking.
Thanks.

**Comment by zarkones** [comment]
Workaround was to use "inTable" instead of "on" method. (Regarding the issue #2)
And for the "this.fn.now()" had to switch to default timestamps. (Issue #1)

**Comment by thetutlage** [comment]
Can you please format your question properly by following this guide https://guides.github.com/features/mastering-markdown/?

**Reply by zarkones** [reply]
I apologize, and have added markups as asked. I hope it's better now. Thanks.

**Comment by thetutlage** [comment, marked as answer]
> Inside up() method I wasn't able to get this.fn.now() to work, in a matter of fact, I failed to obtain Knex reference at all. From which variable may I obtain that reference to Knex variable?

Yes, the `this.fn` property has been removed in favor of `this.now` method. We are trying to keep the interactions with knex transparent and provide the helper methods directly.

> Using on() in context of table.foreign(. . .).references(. . .).on(. . .) I'm prompt with "Property 'on' does not exist on type 'ReferencingColumnBuilder'".

I am not able to find any documentation related to `on`. It has always been `inTable`

**Reply by zarkones** [reply]
For example, this is valid Adonis 4.1 code snippet, but not v5.
`table.foreign(['x', 'y']).references(['id','y']).on('xyz')`

Some small references can be found on: https://adonisjs.com/docs/4.1/database
I was just confused on how it triggers an error in one, but not in the other.
But it worked once when it was replaced with `inTable()`.

Anyhow, `this.now()` works, and thanks for that.
